logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -301,13 +310,77 @@ private void deleteExpiredSegments() throws IOException {
         this.currentQueueSize.set(computeQueueSize());
     }
 
-    private void deleteTailSegment(Path segment, String motivation) throws IOException {
+    /**
+     * Deletes the segment path, if present. Also return the number of events it contains.
+     *
+     * @param segment
+     *      The segment file to delete.
+     * @param motivation
+     *      Description of delete motivation.
+     * @return the number of events contained in the segment or empty optional if the segment was already
+     *      removed.
+     * */
+    private Optional<Long> deleteTailSegment(Path segment, String motivation) throws IOException {
```

Comment:
Somewhere before it escapes the plugin, we should probably catch the `IOException`, and log at `warn` or `error` to avoid cleanup methods crashing the pipeline.

Reply:
Fixed by 5958f0d9bef5d5d98e772e4c0867c6ead87a6996